src/test/ui/const-generics/issues/issue-63322-forbid-dyn.rs

```diff
@@ -0,0 +1,15 @@
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+trait A {}
+struct B;
+impl A for B {}
+
+fn test<const T: &'static dyn A>() {
+    //~^ ERROR the types of const generic parameters must derive `PartialEq` and `Eq`
```

Comment:
The PR seems fine and I'm planning to r+ it, but I wanted to ask about this part:

Do you see the error here as being the absence of `PartialEq` and `Eq`, as indicated by the diagnostic? Or is it that any use of `dyn A` will be broken here?

I personally suspect that, unless we expose a `StructuralMatch` marker trait (at which point maybe one would be able to do `trait A: StructuralMatch {}`), the user error here is in the use of `dyn A`, and therefore the diagnostic message in this case is misleading.

But I'm willing to deal with that in a followup patch.

Reply:
Not sure tbh, I think that we should allow something like `trait A: StructuralMatch {}` in the future.

cc @eddyb i guess